resources/session04/mashup_6.py

```diff
@@ -0,0 +1,263 @@
+    inspection_data = {}
+    use_keys = (
+        'Business Name', 'Average Score', 'Total Inspections', 'High Score'
+    )
+    for key, val in result.items():
+        if key not in use_keys:
+            continue
+        if isinstance(val, list):
+            val = " ".join(val)
+        inspection_data[key] = val
+    sorting_key, marker_type = check_sorting()
+    marker_value = inspection_data.get(sorting_key)
+    if marker_type == 'graduated':
+            inspection_data['marker-color'] = get_color_graduated(marker_value)
+    if marker_type == 'shaded':
+            inspection_data['marker-color'] = get_color_shaded(marker_value)
+    # inspection_data = create_ordered_dict_and_sort(inspection_data, sorting_key)
+    geojson['sort_by'] = marker_value
+    geojson['properties'] = inspection_data
+    # print(geojson)
+    return geojson
+
+
+def get_color_graduated(score):
+    if score >= 90:
+        return '#006400'
+    elif score >= 80:
+        return '#9acd32'
+    elif score >= 70:
+        return '#ffff00'
+    elif score >= 60:
+        return '#ffd700'
+    elif score >= 50:
+        return '#ffa500'
+    elif score >= 40:
+        return '#ff4500'
+    elif score >= 30:
+        return '#ff0000'
+    else:
+        return '#8b0000'
+
+
+def get_color_shaded(score):
+    if score >= 10:
+        return '#000066'
+    elif score >= 8:
+        return '#0000b3'
+    elif score >= 6:
+        return '#0000ff'
+    elif score >= 4:
+        return '#4d4dff'
+    elif score >= 2:
+        return '#b3b3ff'
+    elif score == 1:
+        return '#e5e5ff'
+    else:
+        return '#666699'
+
+
+def check_sorting():
+    marker_type = None
+    if len(argv) >= 2:
+        criteria = argv[1]
+        if criteria == 'average':
+            marker_type = 'graduated'
+            return 'Average Score', marker_type
+        elif criteria == 'highscore' or criteria == 'high_score':
+            marker_type = 'graduated'
+            return 'High Score', marker_type
+        elif criteria == 'inspections' or criteria == 'most_inspections':
+            marker_type = 'shaded'
+            return 'Total Inspections', marker_type
+        else:
+            return criteria, marker_type
+    else:
+        return 'Business Name', marker_type
+
+
+def sort_direction():
+    if len(argv) >= 4:
+        direction = argv[3]
+        if direction == 'reverse':
+            return True
+    else:
+        return False
+
+
+def number_of_listings():
+    if len(argv) >= 3:
+        listings = argv[2]
+        return listings
+    else:
+        return 5
+
+
+if __name__ == '__main__':
+    total_result = {'type': 'FeatureCollection', 'features': []}
+    direction = sort_direction()
+    for result in result_generator(int(number_of_listings())):
+        geojson = get_geojson(result)
+        total_result['features'].append(geojson)
+    total_result['features'] = sorted(total_result.get('features'), key=attrgetter('sort_by'), reverse=direction)
```

Comment:
You are trying to sort items within the generated set of features, but by this time, the number of results has already been narrowed from several hundred to perhaps 10 or 20.  Even if the sorting were working, you'd only be sorting the results you got back, not all of the results, which means your results are invalid.  

As for why the sorting isn't working, the `attrgetter` function from the `operator` module takes a single argument which names the attribute to be gotten, and returns a callable that, when passed an item, will get the named attribute from that item and return it.  This is great when the attribute you want to get is an attribute that is different on all your items, but here you've set the 'sort_by' value to be the same for all items, so sorting on that value will be pointless.  What you really wanted was to pass the name of the sortable key ('Total Inspections', 'High Score', 'Average Score') as the argument to attrgettr (and I think you actually want the `itemgetter` function in any case.  Dictionaries use `__getitem__` to access their contents by key, not `__getattr__`.